core/types/legacy_tx.go

```diff
@@ -379,6 +384,11 @@ func (tx LegacyTx) AsMessage(s Signer, _ *big.Int, _ *chain.Rules) (Message, err
 		rollupDataGas: rollupDataGas(tx),
 	}
 
+	if tx.IsLegacyDepositTx() {
+		msg.from = ZeroAddress
```

Comment:
Is the `from` address actually set to something else? Otherwise, I think this is a no-op?